src/backend/distributed/transaction/backend_data.c

```diff
@@ -487,3 +487,57 @@ AssignDistributedTransactionId(void)
 
 	SpinLockRelease(&MyBackendData->mutex);
 }
+
+
+/*
+ * LockDistributedTransactionSharedMemory locks the shared memory segment
+ * used to keep distributed transaction information to prevent concurrent
+ * reads (e.g. deadlock detection) and writes (e.g. new backend).
+ */
+void
+LockDistributedTransactionSharedMemory(LWLockMode mode)
+{
+	LWLockAcquire(&backendManagementShmemData->lock, mode);
+}
+
+
+/*
+ * UnlockDistributedTransactionSharedMemory releases the light-weight lock
+ * on the shared memory segment for distributed transaction information.
+ */
+void
+UnlockDistributedTransactionSharedMemory(void)
+{
+	LWLockRelease(&backendManagementShmemData->lock);
+}
```

Comment:
For me these wrappers are an abstraction that increases cost at no gains. The cost is small tho...